projects/storefrontapp/src/app/spartacus/features/opf/opf-feature.module.ts

```diff
@@ -25,14 +26,28 @@ if (environment.b2b) {
   extensionProviders.push(provideConfig(defaultOPFCheckoutConfig));
 }
 
+const resultUrl = 'redirect/success';
+const cancelUrl = 'redirect/failure';
```

Comment:
It is essential that resultUrl/CancelUrl values be from the same source in Router and config. Any mismatch would break the PaymentRedirection logic. Thus I also added the routing config (line 25).
The defaultOPFRoutingConfig config is also using resultUrl/CancelUrl values from same source (defaultOpfConfig)
Let me know if you see an alternative solution.